arm64: dts: qcom: msm8916-samsung-rossa: refactor

wonderfulShrineMaidenOfParadise opened the pull request:
Closes #409

- [x] Link: https://lore.kernel.org/r/20251004123907.84270-1-raymondhackley@protonmail.com
- [x] Link: https://lore.kernel.org/all/20260223220514.2556033-1-wonderfulshrinemaidenofparadise@postmarketos.org/

garethppls committed:
arm64: dts: qcom: msm8916-samsung-heatqlte: Add display panel

Signed-off-by: Gareth Peoples <mail@gpeopl.es>
[Raymond: Split from the init commit]
Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

bolilingmeng89006 committed:
arm64: dts: qcom: msm8916-samsung-fortunaltezt: Add display panel

Signed-off-by: "Lin, Meng-Bo" <linmengbo0689@protonmail.com>

wonderfulShrineMaidenOfParadise committed:
squash! arm64: dts: qcom: msm8916-samsung-heatqlte: Add initial device tree

v2: refactor MUIC and touchscreen
Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

wonderfulShrineMaidenOfParadise committed:
squash! arm64: dts: qcom: msm8916-samsung-fortunaltezt/heatqlte: Add PMIC

v2: &charger { richtek,usb-connector = <&usb_con_sm5504>; }; in heatqlte
Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

wonderfulShrineMaidenOfParadise committed:
arm64: dts: qcom: msm8916-samsung-rossa: Add display panel (v2)

v2: move to common device tree
Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

wonderfulShrineMaidenOfParadise committed:
Revert "arm64: dts: qcom: msm8916-samsung-heatqlte: Add display panel"

This reverts commit fa7bb1a3a680d794efab15d2dc4701828a7988e3.

Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

wonderfulShrineMaidenOfParadise committed:
Revert "arm64: dts: qcom: msm8916-samsung-rossa: Add display panel"

This reverts commit 7f1d17e7e1a898f9599f16ca8b7486ebfbc5cd25.

Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-samsung-rossa-common.dtsi:
```diff
@@ -2,26 +2,55 @@
+
+&charger {
+	/* USB connector varies depending on MUIC/extcon */
+	/delete-property/ richtek,usb-connector;
```
How much "code churn" (amount of necessary changes/additional lines) is there if you would omit these properties in fortuna-common and make all fortuna-derived devices specify the MUIC explicitly?

wonderfulShrineMaidenOfParadise committed:
Revert "arm64: dts: qcom: msm8916-samsung-fortunaltezt: Add display panel"

This reverts commit bce07a89ba19b51b7b63ef206a1426e8203b53ba.

Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

bolilingmeng89006 committed:
squash! arm64: dts: qcom: msm8916-samsung-fortunaltezt: Add initial device tree

v2: refactor MUIC
Signed-off-by: "Lin, Meng-Bo" <linmengbo06890@proton.me>

stephan-gh commented on arch/arm64/boot/dts/qcom/msm8916-samsung-coreprimeltevzw.dts:
```diff
@@ -0,0 +1,44 @@
+};
+
+&mpss_mem {
+	/* Firmware for rossa needs more space */
```
```suggestion
	/* Firmware for coreprimeltevzw needs more space */
```

? if every rossa is like that it could be in rossa-common.dtsi

wonderfulShrineMaidenOfParadise replied:
Need to inspect every variant. Will check this later.

wonderfulShrineMaidenOfParadise replied:
```
min_addr: 0x86800000, max_addr: 0x8bc00000, relocatable: True, align: 0x400000
Size: 0x5400000, round: 0x6000000
```

wonderfulShrineMaidenOfParadise committed:
FROMLIST: arm64: dts: qcom: msm8916-samsung-fortuna: Move SM5504 from rossa and refactor MUIC

MUIC varies on fortuna/rossa devices, which could be either SM5502 or
SM5504. Move SM5504 from msm8916-samsung-rossa-common to
msm8916-samsung-fortuna-common and refactor MUIC.

Disable MUIC by default in msm8916-samsung-fortuna-common, and explicitly
specify them in each fortuna/rossa board.

Reviewed-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Link: https://lore.kernel.org/r/20260223220514.2556033-2-wonderfulshrinemaidenofparadise@postmarketos.org
Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

wonderfulShrineMaidenOfParadise committed:
FROMLIST: dt-bindings: qcom: Document samsung,coreprimeltevzw

Document the new samsung,coreprimeltevzw device tree bindings used in
msm8916-samsung-coreprimeltevzw.

Link: https://lore.kernel.org/r/20260223220514.2556033-3-wonderfulshrinemaidenofparadise@postmarketos.org
Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>

chipmunkmc committed:
FROMLIST: arm64: dts: qcom: msm8916-samsung-coreprimeltevzw: add device tree

Samsung Galaxy Core Prime Verizon Wireless is a phone based on MSM8916.
They are similar to the other Samsung devices based on MSM8916 with only a
few minor differences.

The device trees contain initial support with:
 - GPIO keys
 - Regulator haptic
 - SDHCI (internal and external storage)
 - USB Device Mode
 - UART (on USB connector via the SM5502 MUIC)
 - WCNSS (WiFi/BT)
 - Regulators
 - QDSP6 audio
 - Speaker/earpiece/headphones/microphones via digital/analog codec in
   MSM8916/PM8916
 - WWAN Internet via BAM-DMUX
 - PMIC and charger
 - Touchscreen

There are different variants of Core Prime, with some differences in
NFC and MUIC.

The common parts are shared in
msm8916-samsung-fortuna-common.dtsi and msm8916-samsung-rossa-common.dtsi
to reduce duplication.

Signed-off-by: Max McNamee <maxmcnamee@proton.me>
[Raymond: Refactor touchscreen and MUIC. Add commit messages.]
Reviewed-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Link: https://lore.kernel.org/r/20260223220514.2556033-4-wonderfulshrinemaidenofparadise@postmarketos.org
Signed-off-by: Raymond Hackley <raymondhackley@protonmail.com>
